src/include/txn_inline.h

```diff
@@ -251,22 +302,18 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
         WT_ASSERT(session, ref->page != NULL && ref->page->modify != NULL);
         if ((updp = ref->page->modify->inst_updates) != NULL)
             for (; *updp != NULL; ++updp) {
-                (*updp)->start_ts = ts;
-                /*
-                 * Holding the ref locked means we have exclusive access, so if we are committing we
-                 * don't need to use the prepare locked transition state.
-                 */
-                (*updp)->prepare_state = prepare_state;
                 if (commit)
-                    (*updp)->durable_ts = txn->durable_timestamp;
+                    __txn_resolve_prepared_update(session, *updp);
+                else
+                    __txn_prepare_update(session, *updp);
             }
     }
     page_del = ref->page_del;
```

Comment:
Do we need a volatile cast here? This looks like READ_ONCE is needed to me.

Reply:
No, I don't think it is required here. The ref is locked here and other thread can change anything in the ref.